src/listing.rs

```diff
@@ -185,13 +168,18 @@ pub fn directory_listing(
     .to_string();
 
     let breadcrumbs = {
-        let title = title.unwrap_or_else(|| req.connection_info().host().into());
+        let title = conf
+            .title
+            .clone()
+            .unwrap_or_else(|| req.connection_info().host().into());
 
         let decoded = percent_decode_str(&encoded_dir).decode_utf8_lossy();
 
         let mut res: Vec<Breadcrumb> = Vec::new();
-        let mut link_accumulator =
-            format!("/{}", random_route.map(|r| r + "/").unwrap_or_default());
+        let mut link_accumulator = match &conf.random_route {
+            Some(random_route) => format!("/{}/", random_route),
```

Comment:
Yeah, while this is more verbose, I think it's much nicer to read.

Reply:
This was a necessary evil, because otherwise it would be 7 lines long!